### PR TITLE
Add support for LLVM 8

### PR DESCRIPTION
**Pull request by bcardiff**
~~The `IsLocalToUnit`, `IsDefinition`, `IsOptimized` were always used with the same values and the API regarding them changed in llvm 8.0.~~ 

Update: Use `DISubprogram::toSPFlags` to keep functionality & LibLLVMExt API

t - 2 months before llvm 9.0 is released.

### CODE REVIEW

**Comment by bcardiff** [comment]
I've found a better way to implement it via `DISubprogram::toSPFlags` 💯

**Comment by refi64** [comment]
This is actually really sweet, I was working on a Crystal Flatpak SDK a few days ago, but the freedesktop sdk 19.08 beta only ships LLVM 8 so I couldn't use it yet.

File: bin/ci [hunk 1]
```diff
@@ -70,6 +70,7 @@ prepare_system() {
   on_linux sudo service docker restart
 
   on_osx brew update
+  on_osx brew update
```
Comment (bew) [inline bin/ci]: Why 2 times?

Comment (bcardiff) [reply bin/ci]: I'm debugging why llvm@8 is not appearing in CircleCI.

**Comment by anatol** [comment]
Awesome, thank you for porting the codebase to LLVM8.

While we are here, is there any chance to prepare crystal for upcoming LLVM9? Like building the project with LLVM master and see if there are any API changes that can be fixed in advance, before LLVM9 becomes public.

**Comment by bcardiff** [comment]
@anatol There could be preliminary support. Since LLVM 9 is not yet released and there might be pending changes I personally prefer to wait for a release. But definitely, PRs are welcome to have initial support for it if someone wants to tackle it earlier.

**Pull request merged**